module6/interviews/questions-4545.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: page
+title: Interview Questions
+---
+
+You can ask these questions in any order.  There are some questions each week that we know students will not be able to answer - this is by design to see how they exit a tough situation.  Keep track on notes on [this tracker](https://docs.google.com/spreadsheets/d/1eMtwW-06Wv0YAdAnZVm5k8_VFNKEKzQeq24kGxMTHbQ/edit?usp=sharing), please!
+
+
+## Week 1
+* Describe the 4 Pillars of OOP
+* What is HTTP and give a brief definition?
+* Explain Routing in .NET Core MVC?
+* What happens if the inherited interfaces have conflicting method names?
+* What is an enum in C#?
+* Difference between the Equality Operator (==) and Equals() Method in C#
+* Describe your background. How did you get to where you are now? Why coding?
+* Describe your strengths. How have you seen these in practice as a developer?
+* Describe a time when you received feedback that you were surprised by. What did you do with that feedback?
+* Tell me about a successful presentation you gave and why you think it was a hit.
+
+
+## Week 2
+* What is MVC?
+* Explain as much as you can about RESTful APIs
+* What are properties in C#?
+* What are the different data types in JavaScript?
+* What are the Different Ways a Method can be Overloaded?
+* What is Bundling and Minification in MVC?
+* Describe your background. How did you get to where you are now? Why coding?
+* Tell me about a time you had a disagreement with a colleague and how you resolved it.
+* Tell me how you solve a problem.
+* Describe your time at Turing. What worked for you? What didn’t work for you? What was the structure of the program?
+
+## Week 3
+* What are three tools or strategies you use to prevent shipping unstable code to production?
+* What experience do you have working with SQL?
+* What are the types of Joins in SQL?
+* What is LINQ in C#?
+* What is Database First Approach in MVC using Entity Framework?
+* What are Cast() and Convert() Functions in SQL Server?
+* Describe your background. How did you get to where you are now? Why coding?
+* What aspects of software development do you find the most challenging?
+* Describe an example of how you have handled a stressful situation
+* Tell me about a time you failed. How did you deal with the situation?
+
+
+TODO: Update so that students can demonstrate Talking While Coding
```

Comment:
I had originally thought that these interviews would be more leetcode style interviews vs questions. But in hindsight, I think it's good to have both. What do you think about maybe doing the first two weeks question style, and then the last 3 weeks more focused on Leetcode? And we can let folks know that as they are working on freethrows and pair interview practice in the early weeks.

We could even set up a structure where they ask each other these questions during weeks 3-5 in addition to doing a leetcode style interview with an instructor? I like these questions, but I think coding while someone is watching is also an area we could give a lot of helpful feedback in.

Reply:
Definitely - after I put all these questions together, I remembered that we wanted to focus Week 4 on talking while coding 🙃 So, that week I think will definitely be a code challenge.

Week 3, our focus was going to be 'discussing your Turing Experience', so I think that having questions will be give students more chance to show that skill.

What if we switch weeks 3/4 so that we do question, question, code, question, code? That breaks it up some, and let's us do a mix of both on either side of the holiday break?

Reply:
I like that plan! So switching talking while coding and using concrete examples.